streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStateManagerImpl.java

```diff
@@ -198,7 +199,7 @@ public void register(final StateStore store,
         int attempts = 0;
         while (true) {
             try {
-                partitionInfos = globalConsumer.partitionsFor(sourceTopic);
+                partitionInfos = globalConsumer.partitionsFor(sourceTopic, Duration.ofSeconds(30));
```

Comment:
I agree in general, so I'll make that change in the calls to `poll`, but I don't think we have an appropriate config for this call to `partitionsFor`. I just looked up the old call, and it does timeout according to a config. It's just that it waits 305s instead of 30s.

I think I'll revert this line and update the test to set that config lower instead.